diff -urN a/runtime/C/Makefile.am b/runtime/C/Makefile.am
--- a/runtime/C/Makefile.am	2013-07-30 09:46:20.000000000 +0200
+++ b/runtime/C/Makefile.am	2013-07-30 09:52:08.000000000 +0200
@@ -62,7 +62,7 @@
 					include/antlr3treeparser.h				\
 					antlr3config.h
 
-libantlr3c_la_LDFLAGS = -no-undefined -avoid-version
+libantlr3c_la_LDFLAGS = -no-undefined
 libantlr3c_la_LIBADD = $(ANTLR_LIBS)
 
 INCLUDES = -I$(top_srcdir)/include
