diff --git a/build/make/configure.sh b/build/make/configure.sh
index c99a01c..48f8876 100755
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -892,7 +892,7 @@ process_common_toolchain() {
             CXX=${TOOLCHAIN_PATH}/g++
             CC=${TOOLCHAIN_PATH}/gcc
             AR=${TOOLCHAIN_PATH}/ar
-            LD=${TOOLCHAIN_PATH}/arm-apple-darwin10-llvm-gcc-4.2
+            LD=${TOOLCHAIN_PATH}/gcc
             AS=${TOOLCHAIN_PATH}/as
             STRIP=${TOOLCHAIN_PATH}/strip
             NM=${TOOLCHAIN_PATH}/nm
@@ -904,7 +904,7 @@ process_common_toolchain() {
             ASFLAGS="-version -arch ${tgt_isa} -g"
 
             add_cflags -arch ${tgt_isa}
-            add_ldflags -arch_only ${tgt_isa}
+            add_ldflags -arch ${tgt_isa}
 
             if [ -z "${alt_libc}" ]; then
                 alt_libc=${SDK_PATH}/SDKs/iPhoneOS5.1.sdk
