diff -rupN ../srtp_old/crypto/cipher/aes_icm.c ./crypto/cipher/aes_icm.c
--- ../srtp_old/crypto/cipher/aes_icm.c	2006-03-16 18:11:29.000000000 +0100
+++ ./crypto/cipher/aes_icm.c	2011-09-06 10:19:16.000000000 +0200
@@ -281,7 +281,7 @@ aes_icm_set_iv(aes_icm_ctx_t *c, void *i
  * this is an internal, hopefully inlined function
  */
   
-inline void
+static void
 aes_icm_advance_ismacryp(aes_icm_ctx_t *c, uint8_t forIsmacryp) {
   /* fill buffer with new keystream */
   v128_copy(&c->keystream_buffer, &c->counter);
diff -rupN ../srtp_old/crypto/math/datatypes.c ./crypto/math/datatypes.c
--- ../srtp_old/crypto/math/datatypes.c	2005-10-08 18:38:06.000000000 +0200
+++ ./crypto/math/datatypes.c	2011-09-06 10:02:55.000000000 +0200
@@ -124,7 +124,7 @@ octet_string_hex_string(const void *s, i
   return bit_string;
 }
 
-inline int
+static int
 hex_char_to_nibble(uint8_t c) {
   switch(c) {
   case ('0'): return 0x0;
